src/tangent_types/tangent.jl

```diff
@@ -96,6 +96,12 @@ function Base.show(io::IO, tangent::Tangent{P}) where {P}
     end
 end
 
+Base.first(tangent::Tangent{P,T}) where {P,T<:Union{Tuple,NamedTuple}} = first(backing(canonicalize(tangent)))
+Base.last(tangent::Tangent{P,T}) where {P,T<:Union{Tuple,NamedTuple}} = last(backing(canonicalize(tangent)))
+
+Base.tail(tangent::Tangent{P}) where {P<:Tuple} = Tangent{_tailtype(P)}(Base.tail(backing(tangent))...)
```

Comment:
Why is this limited to `Tuple`s only, does `NamedTuple` not work in some way?

It's also not clear to me why `tail` is not defined on any primal `P` (as for `first` and `last`). I thought it is because the order is chosen when defining the `Tangent`, i.e.
```julia
julia> Tangent{Foo}(;a=1.0, b=2.0, c=3.0)
Tangent{Foo}(a = 1.0, b = 2.0, c = 3.0)

julia> Tangent{Foo}(;a=1.0,c=3.0, b=2.0)
Tangent{Foo}(a = 1.0, c = 3.0, b = 2.0)
```
but that would also be an issue for the `last`.

Reply:
I don't know what this should do for Foo. 

`Tangent{Foo}(;a=1.0, b=2.0,)` is very different to what this does. And very different to indexing; `dx[1]` does not produce `Tangent{Foo}(;a=1.0)`. Tangents with wrong-order or wrong-length backing are an aberration, and nothing should deliberately produce them. 

I didn't add NamedTuples just because they would need a more elaborate `_tailtype`.

Reply:
Now NamedTuples are added.